apps/settings/test/unit/panels/call_barring/panel_test.js

```diff
@@ -0,0 +1,620 @@
+    assert.isTrue(this.mockCallBarring.observe.calledWith('boicExhc_enabled'));
+    assert.isTrue(this.mockCallBarring.observe.calledWith('baic_enabled'));
+    assert.isTrue(this.mockCallBarring.observe.calledWith('baicR_enabled'));
+  });
+
+  test('observe updating status before showing >', function() {
+    this.sinon.stub(this.mockCallBarring, 'observe');
+    this.panel.init(document.body);
+    this.panel.beforeShow(document.body);
+    assert.isTrue(this.mockCallBarring.observe.calledWith('updating'));
+  });
+
+  test('stop listening to \'panelready\' before hiding >', function() {
+    this.sinon.stub(window, 'removeEventListener');
+    this.panel.init(document.body);
+    this.panel.beforeHide();
+    assert.isTrue(window.removeEventListener.calledWith('panelready'));
+  });
+
+  test('stop observing settings status before hiding >', function() {
+    this.sinon.stub(this.mockCallBarring, 'unobserve');
+    this.panel.init(document.body);
+    this.panel.beforeHide();
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('baoc'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('boic'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('boicExhc'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('baic'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('baicR'));
+  });
+
+  test('stop observing settings availability before hiding >', function() {
+    this.sinon.stub(this.mockCallBarring, 'unobserve');
+    this.panel.init(document.body);
+    this.panel.beforeHide();
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('baoc_enabled'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('boic_enabled'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith(
+      'boicExhc_enabled'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('baic_enabled'));
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('baicR_enabled'));
+  });
+
+  test('stop observing update status before hiding >', function() {
+    this.sinon.stub(this.mockCallBarring, 'unobserve');
+    this.panel.init(document.body);
+    this.panel.beforeHide();
+    assert.isTrue(this.mockCallBarring.unobserve.calledWith('updating'));
+  });
+
+  test('get data when showing >', function() {
+    this.sinon.stub(this.mockCallBarring, 'getAll');
+    this.panel.init(document.body);
+    this.panel.beforeShow();
+    this.panel.show();
+    assert.isTrue(this.mockCallBarring.getAll.calledWith(_mobileConnection));
+  });
+
+  suite('Click on item, cancel password >', function () {
+    setup(function() {
+      resetHTML();
+      this.panel.init(document.body);
+
+      this.sinon.stub(this.mockPasscode, 'show', function() {
+        return Promise.reject();
+      });
+      this.sinon.spy(this.mockCallBarring, 'set');
+    });
+
+    teardown(function() {
+      document.body.innerHTML = '';
+    });
+
+    test('Barring All Outgoing Calls', function(done) {
```

Comment:
As these tests are exactly the same except for the inputs, can we simply use a for loop to reduce the lines of code?